website_sale_cart_expire/models/website.py

```diff
@@ -37,9 +42,25 @@ def _scheduler_website_expire_cart(self):
             website._get_cart_expire_delay_domain() for website in websites
         ]
         carts_to_expire = self.env["sale.order"].search(
-            expression.OR(carts_to_expire_domains)
+            expression.OR(carts_to_expire_domains), order="id ASC",
         )
         # Expire carts
-        for cart in carts_to_expire:
-            cart.message_post(body=_("Cart expired"))
-        carts_to_expire.action_cancel()
+        with api.Environment.manage():
+            for cart in carts_to_expire:
+                if autocommit:
+                    with registry(self.env.cr.dbname).cursor() as new_cr:
+                        new_env = api.Environment(
```

Comment:
Hm, new cursors + new environments seems a bit excessive now, and redundant with the savepoint

Here's an example of what I meant about autocommit: https://github.com/odoo/odoo/blob/2387366379029d7bf9188e5465cfb4896f11e3b1/addons/event/models/event_mail.py#L169-L184

Naturallty, the cron needs to be changed as well, to pass the autocommit argument, example:
https://github.com/odoo/odoo/blob/2387366379029d7bf9188e5465cfb4896f11e3b1/addons/event/data/event_data.xml#L9